pkg/operator/vsphereproblemdetector/monitoring.go

```diff
@@ -66,6 +70,8 @@ func newMonitoringController(
 		configMapLister:  clients.KubeInformers.InformersFor(csoclients.OperatorNamespace).Core().V1().ConfigMaps().Lister(),
 		eventRecorder:    eventRecorder.WithComponentSuffix("vsphere-monitoring-controller"),
 		monitoringClient: clients.MonitoringClient,
+
+		clusterCSIDriverLister: clients.OperatorInformers.Operator().V1().ClusterCSIDrivers().Lister(),
```

Comment:
You must register the `monitoringController` to get ClusterCSIDriver events below in `WithInformers()`